### PR TITLE
Restructuring + Routing

### PR DESCRIPTION
**Pull request by drewdaemon**

### CODE REVIEW

**Comment by carterworks** [comment]
I get this error when I'm trying to do a `ng build --prod`. Thoughts?

```
ERROR in src/app/pages/login-screen/login-screen.component.html(7,25): : Property 'showRegister' does not exist on type 'LoginScreenComponent'.
src/app/pages/login-screen/login-screen.component.html(19,16): : Property 'showRegister' does not exist on type 'LoginScreenComponent'.
src/app/pages/login-screen/login-screen.component.html(35,16): : Property 'showRegister' does not exist on type 'LoginScreenComponent'.
```

**Comment by drewdaemon** [comment]
@ThatCarterMcBride Pull and check again.

**Comment by ALMerrill** [comment]
Works for me now

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -1,32 +1,48 @@
...
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       declarations: [
         AppComponent,
         CreateGameDialogComponent,
-        GameListComponent
+        GameListComponent,
+        ListedGameComponent,
+        LoginScreenComponent
       ],
```
Comment (ALMerrill) [inline src/app/app.component.spec.ts]: Sorry for the ignorance but is one of these the lobby screen or are we missing that?

File: src/app/app.component.spec.ts [hunk 2]
```diff
@@ -1,32 +1,48 @@
...
+import { ListedGameComponent } from './pages/game-list/listed-game/listed-game.component';
+import { CreateGameDialogComponent } from './pages/game-list/create-game-dialog/create-game-dialog.component';
+import { LoginScreenComponent } from './pages';
+
+import { RouterModule, Routes } from '@angular/router';
+
+const appRoutes: Routes = [
+  { path: 'login', component: LoginScreenComponent },
+  { path: 'game-list', component: GameListComponent },
+  {
```
Comment (ALMerrill) [inline src/app/app.component.spec.ts]: Are we just not redirecting to the lobby screen?

**Review by carterworks — APPROVED** [review]
:+1:

**Comment by carterworks** [comment]
@drewctate We are merging this PR so @bandrus5 can use it in his PR

**Pull request merged**